### ISSUE TITLE
Manage Endpoints for Campaign

### ISSUE THREAD

**Issue by stevejgordon** [HTBox/allReady#1317]
Once we add features to update getasmokealarm as status changes occur in requests, do we need a way to flag a campaign as having a service endpoint?

While we have a single campaign this shouldn't be an issue but in the future we wouldn't want to try and update getasmokealarm each time a status changes on a request for a different campaign.

**Comment by tonysurma** [comment]
Yes we need some way to flag this and tie them.  cc @MisterJames @mgmccarthy

**Comment by stevejgordon** [comment]
I'll grab this one and try to add something basic to get us going. I'm thinking a table to hold details of the service integrations we support. Each integration would be tied to an Org.

e.g.
Id - 1
Name - GetASmokeAlarm
Org - Red Cross

Then, when creating a campaign an Org Admin can select from any of their available integrations (or leave empty) for the new campaign.

We would then use this integration information wherever we may need to make integration calls. A basic approach would be to check what, if any integrations we need to work with. This could get quite complicated as the GASA is quite tied to our flow right now.

Just opening up for thoughts before I start working on this.

/cc @mgmccarthy

**Comment by mgmccarthy** [comment]
I agree with a higher-level approach like this... 

where this starts to get tricky is how we end up abstraction these kind of details in code... for example, we have the `RequestApiController` which has a `Post` method which takes a `RequestApiViewModel`. This view-model is ALMOST vendor agnostic... the one piece of data that ruins it is the `ProviderData` field, which is really something that was added for GASA.

What I'm envisioning is using RequestApiController as a "router" of sorts... we can check the data we'll be keeping in that @stevejgordon talked about in his [comment above](https://github.com/HTBox/allReady/issues/1317#issuecomment-266538651), and then based on that information, route the incoming API request to a specific controller/action, or multiple actions in the same controller.

The whole point is we'll need to somehow allow the addition of other information if we stick with a single endpoint, as the ViewModel should be the "base" of the data we need.

Other vendor-specific data/scenarios should not be tacked onto `RequestApiViewModel` and should not be stored on the `Request` model.

The `Request` model needs to stay focused on how we internally represent and operate with Requests in our system. If we need to keep other vendor-specific data around as a result of the incoming request, we'll need to create a new table or tables that will allow us to include information on reads for dealing with that specific vendor implementation.

We've already "polluted" the `Request` model with the `ProviderData` field, as that has no bearing on how we internally operate with a Request. Either way, we'll have to figure these details out per integration.

There are a couple different ways we can go here, but starting with what @stevejgordon proposed above is a good way to do it. Let's focus on the user experience and functionally, how we want to let admins add/change existing inbound integrations, and once we identify that, we can move forward with how we can build some reusable pieces of the code, and abstract vendor-specific details away from that main, reusable code.

**Comment by mgmccarthy** [comment]
questions for @tonysurma @mikesigs @OhMcGoo

is it possible for one integration (let's say the importing of Requests right now) to be used across multiple organizations? I'm assuming the answer to this is yes, but I was just chatting with @stevejgordon about it.

**Comment by tonysurma** [comment]
The answer is yes.  For v1 we can assume it is only for one org, but over time yes it definitely is possible.
